src/component/legend/LegendView.ts

```diff
@@ -564,8 +568,14 @@ function getLegendStyle(
     const legendItemModel = legendModel.getModel('itemStyle') as Model<LegendItemStyleOption>;
     const itemStyle = legendItemModel.getItemStyle();
     const iconBrushType = iconType.lastIndexOf('empty', 0) === 0 ? 'fill' : 'stroke';
+    const decalStyle = legendItemModel.getShallow('decal', false);
+    if (!decalStyle || decalStyle === 'inherit') {
```

Comment:
I will prefer using a ternary expression here for less code:
```ts
itemStyle.decal = !decalStyle || decalStyle === 'inherit' 
  ? itemVisualStyle.decal
  : createOrUpdatePatternFromDecal(decalStyle, api)

Reply:
For sure.